libs/dh/admin/feature-user-management/src/lib/feature-create-userrole/tabs/dh-create-userrole-tabs.component.ts

```diff
@@ -0,0 +1,196 @@
+
+interface CreateRoleForm {
+  masterData?: ObservedValueOf<
+    DhCreateUserroleMasterdataTabComponent['formReady']
+  >;
+}
+@Component({
+  selector: 'dh-create-userrole-tabs',
+  standalone: true,
+  templateUrl: './dh-create-userrole-tabs.component.html',
+  providers: [
+    provideComponentStore(DhAdminUserRolesManagementDataAccessApiStore),
+  ],
+  styles: [
+    `
+      :host {
+        display: block;
+      }
+    `,
+  ],
+  imports: [
+    TranslocoModule,
+    CommonModule,
+    PushModule,
+    LetModule,
+    WattTabsModule,
+    WattValidationMessageModule,
+    DhCreateUserroleMasterdataTabComponent,
+    DhCreateUserrolePermissionsTabComponent,
+    WattCardModule,
+    WattSpinnerModule,
+    WattButtonModule,
+    ReactiveFormsModule,
+    FormsModule,
+  ],
+})
+export class DhCreateUserroleTabsComponent implements OnInit, OnDestroy {
+  private readonly store = inject(DhAdminUserRolesManagementDataAccessApiStore);
+  isLoading$ = this.store.isLoading$;
+  validation$ = this.store.validation$;
+  createRoleForm = this.formBuilder.group<CreateRoleForm>({});
+  userRole: CreateUserRoleDto;
+  selectablePermissions$ = this.store.selectablePermissions$;
+
+  private destroy$ = new Subject<void>();
+
+  constructor(
+    private router: Router,
+    private formBuilder: FormBuilder,
+    private toastService: WattToastService,
+    private translocoService: TranslocoService
+  ) {
+    this.userRole = {
+      name: '',
+      description: '',
+      eicFunction: 'Agent',
+      status: 'Active',
+      permissions: [],
+    };
+  }
+
+  ngOnInit(): void {
+    this.store.createRequestHasError$
+      .pipe(takeUntil(this.destroy$))
+      .subscribe((hasError) => {
+        if (hasError) {
+          this.toastService.open({
+            message: this.translocoService.translate(
+              'admin.userManagement.createrole.createRoleRequest.error'
+            ),
+            type: 'danger',
+          });
+
+          this.enableControls();
+        }
+      });
+  }
+
+  ngOnDestroy(): void {
+    this.destroy$.next();
+    this.destroy$.unsubscribe();
+  }
+
+  onEicFunctionSelected = (eic: EicFunction) => {
+    console.log(eic);
+  };
+
+  onSubmit() {
+    if (!this.userRole) throw new Error('Missing user role');
+    this.disableControls();
+    this.store.createUserRole({
+      createRole: this.userRole,
+      onSaveCompletedFn: this.backToOverviewAfterSave,
+    });
+
+    this.toastService.open({
+      message: this.translocoService.translate(
+        'admin.userManagement.createrole.createRoleRequest.start'
+      ),
+      type: 'loading',
+    });
+  }
+
+  disableControls() {
+    this.createRoleForm.controls.masterData?.controls.description.disable();
+    this.createRoleForm.controls.masterData?.controls.eicFunction.disable();
+    this.createRoleForm.controls.masterData?.controls.name.disable();
+    this.createRoleForm.controls.masterData?.controls.roleStatus.disable();
```

Comment:
suggestion: since all `FormControl`s are grouped within `createRoleForm`, we can probably do `this.createRoleForm.disable()` to disable all of them, instead of doing it one by one.

Reply:
I'll test and adjust if successfull

Reply:
Worked, so updated to this instead.